internal/providers/terraform/aws/elastic_beanstalk_environment.go

```diff
@@ -0,0 +1,89 @@
+package aws
+
+import (
+	"github.com/infracost/infracost/internal/resources/aws"
+	"github.com/infracost/infracost/internal/schema"
+)
+
+func getElasticBeanstalkEnvironmentRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_elastic_beanstalk_environment",
+		RFunc: newElasticBeanstalkEnvironment,
+	}
+}
+
+func newElasticBeanstalkEnvironment(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+
+	var region = d.Get("region").String()
+
+	r := &aws.ElasticBeanstalkEnvironment{
+		Address: d.Address,
+		Region:  region,
+		Name:    d.Get("name").String(),
+	}
+
+	lc := &aws.LaunchConfiguration{
+		Address: "aws_launch_configuration",
+		Region:  region,
+		Tenancy: "dedicated",
+	}
+	volume := &aws.EBSVolume{
+		Address: "aws_ebs_volume",
+		Region:  region,
+	}
+	db := &aws.DBInstance{
+		Address: "aws_db_instance",
+		Region:  region,
+	}
+	cwlg := &aws.CloudwatchLogGroup{
+		Address: "aws_cloudwatch_log_group",
+		Region:  region,
+	}
+	elb := &aws.ELB{
+		Address: "aws_elb",
+		Region:  region,
+	}
+	lb := &aws.LB{
+		Address: "aws_loadbalancer",
+		Region:  region,
+	}
+
+	for _, setting := range d.Get("setting").Array() {
+		switch setting.Get("name").String() {
+		case "InstanceTypes":
+			lc.InstanceType = setting.Get("value").String()
+		case "MinSize":
+			lc.InstanceCount = intPtr(setting.Get("value").Int())
+		case "RootVolumeSize":
+			volume.Size = intPtr(setting.Get("value").Int())
+		case "RootVolumeIOPS":
+			volume.IOPS = setting.Get("value").Int()
+		case "RootVolumeType":
+			volume.Type = setting.Get("value").String()
+		case "LoadBalancerType":
+			r.LoadBalancerType = setting.Get("value").String()
+		case "StreamLogs":
+			r.StreamLogs = true
+		case "DBInstanceClass":
+			r.RDSIncluded = true
```

Comment:
I'd move this to a `var` at this level. 

Then line `80` can go to this:

```go
if dbIncluded {
	r.DBInstance = db
}
```

This means that in `ElasticBeanstalkEnvironment` you can remove the properties and just check if `DBInstance` is not nil.

e.g:

```go
if r.DBInstance != nil {
    resources.PopulateArgsWithUsage(r.DBInstance, u)
}
```